src/mixins/themeable.ts

```diff
@@ -39,108 +51,170 @@ export interface ThemeableOptions {
-		const previousThemeClasses = themeClassesMap.get(instance);
-		themeClassesMap.set(instance, negatePreviousClasses(previousThemeClasses, newThemeClasses));
-	} else {
-		themeClassesMap.set(instance, newThemeClasses);
-	}
+		allClasses = [...allClasses, ...cssClassNames];
+
+		newAppliedClassNames[className] = setClassNameApplied(cssClassNames, true);
+
+		return newAppliedClassNames;
+	}, <CSSModuleClasses> {});
+
+	appendToAllClassNames(instance, allClasses);
+
+	return themeClasses;
 }
 
-function onPropertiesChanged<T>(instance: Themeable<T>, { theme, overrideClasses }: ThemeableProperties, changedPropertyKeys: string[]) {
+function onPropertiesChanged(instance: Themeable, { theme, overrideClasses }: ThemeableProperties, changedPropertyKeys: string[]) {
 	const themeChanged = includes(changedPropertyKeys, 'theme');
 	const overrideClassesChanged = includes(changedPropertyKeys, 'overrideClasses');
 
 	if (themeChanged || overrideClassesChanged) {
 		const themeClasses = generateThemeClasses(instance, instance.baseTheme, theme, overrideClasses);
-		updateThemeClassesMap(instance, themeClasses);
+		cssModuleClassNameMap.set(instance, themeClasses);
 	}
 }
 
+function createBaseThemeLookup({ classes }: BaseTheme): ClassNames {
+	return Object.keys(classes).reduce((currentClassNames, key: string) => {
+		currentClassNames[classes[key]] = key;
+		return currentClassNames;
+	}, <ClassNames> {});
+}
+
+function splitClassStrings(...classes: string[]): string[] {
+	return classes.reduce((splitClasses: string[], className) => {
+		if (className.indexOf(' ') > -1) {
+			splitClasses.push(...className.split(' '));
+		} else {
+			splitClasses.push(className);
+		}
+		return splitClasses;
+	}, []);
+}
+
 /**
  * Themeable Factory
  */
 const themeableFactory: ThemeableFactory = createEvented.mixin({
 	mixin: {
-		get theme(this: Themeable<any>): AppliedClasses<any> {
-			return themeClassesMap.get(this);
+		classes(this: Themeable, ...classNames: string[]) {
+			const cssModuleClassNames = cssModuleClassNameMap.get(this);
+			const baseThemeLookup = baseThemeLookupMap.get(this);
+
+			const appliedClasses = classNames.reduce((currentCSSModuleClassNames, className) => {
+				const classNameKey = baseThemeLookup[className];
+				if (cssModuleClassNames.hasOwnProperty(classNameKey)) {
+					assign(currentCSSModuleClassNames, cssModuleClassNames[classNameKey]);
+				} else {
+					console.error(`Class name: ${className} and lookup key: ${classNameKey} not from baseTheme, use chained 'fixed' method instead`);
+				}
+				return currentCSSModuleClassNames;
+			}, <any> {});
+
+			let responseClasses = assign({}, allClassNamesMap.get(this), appliedClasses);
+			const instance = this;
+
+			const classesResponseChain: ClassesChain = {
+				classes: responseClasses,
+				fixed(this: ClassesChain, ...classes: string[]) {
+					const splitClasses = splitClassStrings(...classes);
+					assign(this.classes, setClassNameApplied(splitClasses, true));
```

Comment:
I know this isn't where the function is defined but i am not keen on `setClassNameApplied`, because I think it actually creates the object from a string of keys (unless i am mistaken).